src/fluent-plugin-mdsd/lib/fluent/plugin/out_mdsd.rb

```diff
@@ -51,18 +53,22 @@ def shutdown()
         end
 
         # This method is called when an event reaches to Fluentd.
+        # time: this is an integer, Unix time_t. Number of seconds since 1/1/1970 UTC.
         # NOTE: a plugin must define this because base class doesn't have
         # default implementation.
         def format(tag, time, record)
-            [tag, record].to_msgpack
+            [tag, time, record].to_msgpack
         end
 
         # This method is called every flush interval.
         #
         # NOTE! This method is called by internal thread, not Fluentd's main thread.
         # So IO wait doesn't affect other plugins.
         def write(chunk)
-            chunk.msgpack_each {|(tag, record)|
+            chunk.msgpack_each {|(tag, time, record)|
+                # Ruby (version >= 1.9) hash preserves insertion order. So the following item is
+                # the last item when iterating the 'record' hash.
```

Comment:
this is in the Ruby spec. e.g. [http://ruby-doc.org/core-2.3.0/Hash.html](http://ruby-doc.org/core-2.3.0/Hash.html).  "Hashes enumerate their values in the order that the corresponding keys were inserted."

---
In reply to: [114448777](https://github.com/Azure/fluentd-plugin-mdsd/pull/28#discussion_r114448777) [](ancestors = 114448777)